### PR TITLE
Add before_emit_metric callback to options

### PR DESCRIPTION
**Pull request by markushi**
## :pencil: Checklist

- [ ] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [ ] No breaking change or entry added to the changelog.
- [ ] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: sentry/src/test/java/io/sentry/SentryOptionsTest.kt [hunk 1]
```diff
@@ -651,6 +652,15 @@ class SentryOptionsTest {
         assertFalse(options.isEnableSpanLocalMetricAggregation)
     }
 
+    @Test
+    fun `when metric callback is set, getter returns it`() {
+        val callback = SentryOptions.BeforeEmitMetricCallback { _, _ -> false }
+        val options = SentryOptions().apply {
+            beforeEmitMetricCallback = callback
+        }
+        assertSame(options.beforeEmitMetricCallback, options.beforeEmitMetricCallback)
```
Comment (stefanosiano) [inline sentry/src/test/java/io/sentry/SentryOptionsTest.kt:661]:
```suggestion
        assertSame(callback, options.beforeEmitMetricCallback)
```

File: sentry/src/test/java/io/sentry/SentryOptionsTest.kt [hunk 2]
```diff
@@ -651,6 +652,15 @@ class SentryOptionsTest {
...
+    }
+
```
Comment (stefanosiano) [inline sentry/src/test/java/io/sentry/SentryOptionsTest.kt:663]: add a test for default value of `beforeEmitMetricCallback` null?

**Review by stefanosiano — APPROVED** [review]
just minor detail in tests, but good!
